File: Data Structures & Algorithms/duplicate-integer/submission-3.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        unique_nums = set()
        for i in range(len(nums)):
            if nums[i] in unique_nums:
                return True
            else:
                unique_nums.add(nums[i])
        return False


        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int[] sorted = nums;
        Arrays.sort(sorted);
        int res = 1;
        int cur = 1;

        if (nums.length == 0) return 0;

        for (int i = 1; i < nums.length; i++){
            if (Math.abs(sorted[i] - sorted[i-1]) == 0){
                continue;
            }
            else if (Math.abs(sorted[i] - sorted[i-1]) == 1){
                cur++; 
            }
            else {
                cur = 1; 
            }
            if (cur > res){
                    res = cur;
                }

        }
        return res;
    }
}


